builtin/providers/ultradns/resource_ultradns_record.go

```diff
@@ -0,0 +1,181 @@
+package ultradns
+
+import (
+	"fmt"
+	"github.com/Ensighten/udnssdk"
+	"github.com/hashicorp/terraform/helper/schema"
+	"log"
+	"strconv"
+)
+
+func resourceUltraDNSRecord() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceUltraDNSRecordCreate,
+		Read:   resourceUltraDNSRecordRead,
+		Update: resourceUltraDNSRecordUpdate,
+		Delete: resourceUltraDNSRecordDelete,
+
+		Schema: map[string]*schema.Schema{
+			"zone": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"hostname": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"rdata": &schema.Schema{
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"ttl": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "3600",
+			},
+		},
+	}
+}
+
+func resourceUltraDNSRecordCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*udnssdk.Client)
+	newRecord := &udnssdk.RRSet{
+		OwnerName: d.Get("name").(string),
+		RRType:    d.Get("type").(string),
+	}
+	rdata := d.Get("rdata").([]interface{})
+	rdatas := make([]string, len(rdata))
+	for i, j := range rdata {
+		rdatas[i] = j.(string)
+	}
+	newRecord.RData = rdatas
+	ttl := d.Get("ttl").(string)
+	newRecord.TTL, _ = strconv.Atoi(ttl)
+
+	log.Printf("[DEBUG] UltraDNS RRSet create configuration: %#v", newRecord)
+
+	_, err := client.RRSets.CreateRRSet(d.Get("zone").(string), *newRecord)
+	recId := fmt.Sprintf("%s.%s", d.Get("name").(string), d.Get("zone").(string))
+	if err != nil {
+		return fmt.Errorf("Failed to create UltraDNS RRSet: %s", err)
+	}
+
+	d.SetId(recId)
+	log.Printf("[INFO] record ID: %s", d.Id())
+
+	return resourceUltraDNSRecordRead(d, meta)
+}
+
+func resourceUltraDNSRecordRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*udnssdk.Client)
+
+	rrsets, _, err := client.RRSets.GetRRSets(d.Get("zone").(string), d.Get("name").(string), d.Get("type").(string))
+	if err != nil {
+		uderr, ok := err.(*udnssdk.ErrorResponseList)
+		if ok {
+			for _, r := range uderr.Responses {
+				// 70002 means Records Not Found
+				if r.ErrorCode == 70002 {
+					d.SetId("")
+					return nil
+				} else {
+					return fmt.Errorf("Couldn't find UltraDNS RRSet: %s", err)
+				}
+			}
+		} else {
+			return fmt.Errorf("Couldn't find UltraDNS RRSet: %s", err)
+		}
+	}
+	rec := rrsets[0]
+	err = d.Set("rdata", rec.RData)
+	if err != nil {
+		return fmt.Errorf("[DEBUG] Error setting records: %#v", err)
+	}
+	d.Set("ttl", rec.TTL)
+
+	if rec.OwnerName == "" {
+		d.Set("hostname", d.Get("zone").(string))
+	} else {
+		if string(rec.OwnerName[len(rec.OwnerName)-1]) == "." {
```

Comment:
[`strings.HasSuffix`](https://golang.org/pkg/strings/#HasSuffix) should work nicely here